test/e2e/manifests/pkg/deployment-runtime-config/setup/deployment-runtime-config.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: pkg.crossplane.io/v1beta1
+kind: DeploymentRuntimeConfig
+metadata:
+  name: function-dummy-rc
+spec:
+  serviceAccountTemplate:
+    metadata:
+      name: ibelieveicanfly # we can the service account name here and expect it to be used in the deployment even if we don't specify it in the deployment template
+      labels:
+        foo: bar
+  serviceTemplate:
+    metadata:
+      name: letscomplicateitfurther # this will require tls certificates to be generated properly for this service name
+  deploymentTemplate:
+    metadata:
+      name: iamfreetochoose
+    spec:
+      replicas: 2
+      selector: {}
+      template:
+        #metadata: # This is rejected with unknown field "spec.deploymentTemplate.spec.template.metadata.labels"
+        #  labels:
+        #    somepodlabels: cool
```

Comment:
Is this meant to be here?

Reply:
I'll eventually clean it up, but I would like to ask for some help here. 

I couldn't understand why controller tools didn't generate `metadata.labels` in the schema of `DeploymentRuntimeConfig` and would appreciate some help here. I don't want to ignore this, as this is how one could add labels and annotations on the pod instead of the deployment. Any ideas?

<img width="814" alt="Screenshot 2023-10-16 at 15 04 10" src="https://github.com/crossplane/crossplane/assets/9900707/cf3dfb55-db56-40a2-b33f-2addbf57e458">

Reply:
Is it possible controller-tools has special handling for types named ObjectMeta?

I believe you don't need to specify CRD OpenAPI schema for ObjectMeta when it's in the "usual place" (i.e. top-level `metadata`) because the API server handles that automatically. Perhaps controller-tools has some logic to handle that that is affecting us.

Reply:
Oh right, I just realized this is the deployment template's pod template's metadata. It looks like you can enable generation of this using a controller-runtime argument. See https://github.com/crossplane/crossplane/commit/f7e1ddf4f0782b1e5594ebacb077d3416a55db84

Reply:
Great, thanks a lot!